internal/services/dedicatedcloud/dedicatedcloud.go

```diff
@@ -6,23 +6,649 @@ package dedicatedcloud
 
 import (
 	_ "embed"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
 
+	"github.com/ovh/ovhcloud-cli/internal/display"
+	"github.com/ovh/ovhcloud-cli/internal/filters"
 	"github.com/ovh/ovhcloud-cli/internal/flags"
-	"github.com/ovh/ovhcloud-cli/internal/services/common"
+	httpLib "github.com/ovh/ovhcloud-cli/internal/http"
 	"github.com/spf13/cobra"
 )
 
 var (
-	dedicatedcloudColumnsToDisplay = []string{"serviceName", "location", "state", "description"}
+	dedicatedcloudColumnsToDisplay = []string{"serviceName", "regionLocation", "version", "state", "description"}
 
 	//go:embed templates/dedicatedcloud.tmpl
 	dedicatedcloudTemplate string
+
+	//go:embed templates/datacenter.tmpl
+	datacenterTemplate string
 )
 
+// toFloat64 converts any numeric type to float64
+func toFloat64(v any) float64 {
+	switch n := v.(type) {
+	case float64:
+		return n
+	case int:
+		return float64(n)
+	case json.Number:
+		if f, err := n.Float64(); err == nil {
+			return f
+		}
+	}
+	return 0
+}
+
+// toInt converts any numeric type to int
+func toInt(v any) int {
+	switch n := v.(type) {
+	case int:
+		return n
+	case float64:
+		return int(n)
+	case json.Number:
+		if i, err := n.Int64(); err == nil {
+			return int(i)
+		}
+	}
+	return 0
+}
+
+// getLocationMap fetches the location mapping from the API
+func getLocationMap() (map[string]string, error) {
+	// Fetch location list (array of location IDs)
+	var locationIDs []string
+	if err := httpLib.Client.Get("/dedicatedCloud/location", &locationIDs); err != nil {
+		return nil, fmt.Errorf("failed to fetch location list: %w", err)
+	}
+
+	// Build location map by fetching each location detail
+	locationMap := make(map[string]string)
+	for _, locationID := range locationIDs {
```

Comment:
you should use `httpLib.FetchObjectsParallel` instead (https://github.com/ovh/ovhcloud-cli/blob/main/internal/http/client.go#L50)